### PR TITLE
Introduce automation email type [MAILPOET-4432]

### PR DESCRIPTION
**Pull request by websupporter**
Fixes [MAILPOET-4432]

This PR introduces a new email type "automation".
I had a look for more tests, which might need extension but didn't really find a lot.

[MAILPOET-4432]: https://mailpoet.atlassian.net/browse/MAILPOET-4432?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: mailpoet/tests/integration/API/JSON/v1/SendingQueueTest.php [hunk 1]
```diff
@@ -43,6 +43,10 @@ public function _before() {
     ]);
   }
 
+  public function testItDoesNotScheduleAutomationEmailsWithTheSendingQueue() {
```
Comment (alex-mailpoet) [inline mailpoet/tests/integration/API/JSON/v1/SendingQueueTest.php]: Please remove or implement this empty test.

**Review by alex-mailpoet — COMMENTED** [review]
I have a single tiny comment, otherwise it's good.

File: mailpoet/tests/integration/API/JSON/v1/SendingQueueTest.php (hunk 1 above)
Comment (websupporter) [reply mailpoet/tests/integration/API/JSON/v1/SendingQueueTest.php]: Thanks for catching @alex-mailpoet. I removed the test.

**Review by alex-mailpoet — APPROVED** [review]

**Comment by alex-mailpoet** [comment]
We decided to merge it because we need to build on these changes. They shouldn't affect existing functionality.

**Pull request merged**